uplink/storage/streams/store.go

```diff
@@ -298,15 +298,52 @@ func (s *streamStore) Get(ctx context.Context, path Path, pathCipher storj.Ciphe
 		return nil, Meta{}, err
 	}
 
-	object, err := s.metainfo.GetObject(ctx, metainfo.GetObjectParams{
-		Bucket:        []byte(path.Bucket()),
-		EncryptedPath: []byte(encPath.Raw()),
-	})
+	resps, err := s.metainfo.Batch(ctx,
+		&metainfo.GetObjectParams{
+			Bucket:        []byte(path.Bucket()),
+			EncryptedPath: []byte(encPath.Raw()),
+		},
+		&metainfo.DownloadSegmentParams{
+			Position: storj.SegmentPosition{
+				Index: -1,
```

Comment:
Can we add a comment on what downloading segment `-1` means?

Reply:
That's a good idea for improving what it was there.
If you know clearly what it means let me know otherwise I'll investigate a bit because I have some idea but I don't want to write an assumption of something that I'm not 100%.

Reply:
It means `lastSegment`

Reply:
It means `lastSegment`

Reply:
Requesting for last segment seems weird, we should be trying to get slowly rid of the concept on uplink side to ensure that multipart uploads won't break old uplinks.

Reply:
@egonelbre unless that we could accommodate that in a few lines and you can tell me what to do, let's open a ticket please, this is a refactoring to do batch requests.

The same `-1` was requested with a single request.

Reply:
Sure that's fine, for now... but giving a heads up.

I think the batch request should be something like:
```
get object params
list segments
download segment params (implicitly carried over from the list)
```

Of course, I'm not sure whether it would work right now.

Reply:
It's not possible, for now, only `segmentID` and  `streamID` are propagated https://storjlabs.atlassian.net/browse/V3-2617

I'll open a ticket